src/components/AddNote.js

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import { useEffect, useState } from "react";
+import axios from "axios";
+import "./NoteCard.css";
+import { useNotes } from "../backend/context/NotesContext";
+
+const AddNote = () => {
+  const { encodedToken, setCounter } = useNotes();
+  const [noteContent, setNoteContent] = useState({
+    title: "",
+    body: "",
+  });
+  const [txtareaHeight, setTxtareaHeight] = useState(1);
+
+  const addCardTitle = (e) => {
+    setNoteContent((i) => ({ ...i, title: e.target.value }));
+  };
+  const addCardBody = (e) => {
+    setNoteContent((i) => ({ ...i, body: e.target.value }));
+    setTxtareaHeight(e.target.value.length / 50 + 2);
+  };
+  const saveClickHandler = async () => {
+    setNoteContent((i) => ({ ...i, title: "", body: "" }));
+    setCounter((counter) => counter + 1);
+    console.log(noteContent);
+    try {
+      const noteRes = await axios.post(
+        "/api/notes",
+        {
+          note: noteContent,
+        },
+        {
+          headers: {
+            authorization: encodedToken,
+          },
+        }
+      );
+    } catch (err) {
+      console.log(err);
+    }
+  };
```

Comment:
You can define this function in the NotesContext.js and call it here through the useNotes Hook because it is good to keep all the logic and API call operations in the context and the components must be there only for viewing or rendering the data related to it and display the behaviour on different events.